protocol/bc/txinput.go

```diff
@@ -63,10 +62,10 @@ func NewSpendInput(txhash Hash, index uint32, arguments [][]byte, assetID AssetI
 		ControlProgram: controlProgram,
 	}
 	return &TxInput{
-		AssetVersion:  av,
+		AssetVersion:  assetVersion,
```

Comment:
I still prefer a literal here. This function produces a version 1 txinput only, not whatever kind of txinput `assetVersion` is defined to be in the future.